src/AutoRest.CSharp/Mgmt/Decorator/TypeReferenceTypeChooser.cs

```diff
@@ -42,6 +42,12 @@ internal static class TypeReferenceTypeChooser
 
             foreach (System.Type replacementType in TypeReferenceTypes)
             {
+                // DataFactory reference types should only be used to replace generated types in a DataFactory namespace.
```

Comment:
This was necessary to avoid false positive replacements. If we need a more robust attribute-based solution we can investigate that as well, but this should at least unblock the DataFactory release.